src/zenml/cli/stack_recipes.py

```diff
@@ -0,0 +1,806 @@
+        """Create a new LocalStack instance.
+
+        Args:
+            name: The name of the stack, specifically the name of the folder
+                  on git
+            path: Path at which the stack is installed
+        """
+        self.name = name
+        self.path = path
+
+    @property
+    def recipes_deploy_bash_script(self) -> str:
+        """Path to the bash script that deploys the recipe.
+
+        Returns:
+            Path to the bash script that deploys the recipe.
+        """
+        return os.path.join(self.path, RECIPE_RUN_SCRIPT)
+
+    @property
+    def recipes_destroy_bash_script(self) -> str:
+        """Path to the bash script that destroys the recipe.
+
+        Returns:
+            Path to the bash script that destroys the recipe.
+        """
+        return os.path.join(self.path, RECIPE_DESTROY_SCRIPT)
+
+    def is_present(self) -> bool:
+        """Checks if the stack_recipe exists at the given path.
+
+        Returns:
+            True if the stack_recipe exists at the given path, else False.
+        """
+        return fileio.isdir(str(self.path))
+
+    def run_stack_recipe(
+        self, stack_recipe_runner: List[str], identifier: str
+    ) -> None:
+        """Run the local stack recipe using the bash script at the supplied location.
+
+        Args:
+            stack_recipe_runner: Sequence of locations of executable file(s)
+                            to run the stack_recipe
+            identifier: String to identify if deploy or destroy is being run
+
+        Raises:
+            NotImplementedError: If the stack_recipe hasn't been implement yet.
+            FileNotFoundError: If the stack_recipe runner script is not found.
+            subprocess.CalledProcessError: If the stack_recipe runner script fails.
+        """
+        if all(map(fileio.exists, stack_recipe_runner)):
+            call = stack_recipe_runner
+            try:
+                subprocess.check_call(
+                    call,
+                    cwd=str(self.path),
+                    shell=click._compat.WIN,
+                    env=os.environ.copy(),
+                )
+            except RuntimeError:
+                raise NotImplementedError(
+                    f"Currently the recipe {self.name} "
+                    "has no implementation for the "
+                    f"{identifier} method"
+                )
+            except subprocess.CalledProcessError as e:
+                if e.returncode == 38:
+                    raise NotImplementedError(
+                        f"Currently the recipe {self.name} "
+                        "has no implementation for the "
+                        f"{identifier} method"
+                    )
+                raise
+        else:
+            raise FileNotFoundError(
+                "Bash File(s) to run recipes not found at"
+                f"{stack_recipe_runner}"
+            )
+
+        # Telemetry
+        if identifier == "deploy":
```

Comment:
I'm not a fan of passing in some `identifier` as a string here to be honest. I think making this an enum like `RecipeExecutionMode` would be a little nicer, but seeing as this is only used to track analytics, I don't see any reason to even pass this and not call `track_event` in the `deploy` and `destroy` CLI functions below?

Reply:
The identifier isn't passed as a string but by using a constant value. I agree that the string shouldn't be used in the comparison here. How would you feel if I made the following change?
```py
if identifier == RECIPE_DEPLOY_IDENTIFIER
```
The identifier is also used to make error messages more friendly and specific above as this function is called by both deploy and destroy functions.

Reply:
Yes that would be similar to the enum approach, definitely better than the previous solution and fine with me.

I think passing the identifier could still be avoided by doing something like this:
```python
def deploy(recipe_name):
    try:
        local_stack_recipe.run_stack_recipe(...)
    except SomeError as e:
        cli_utils(f"Failed to deploy recipe {recipe_name}. Reason: {e}")
```
In any case, as long as the string constants are gone I can live with it :)

Reply:
That's a really elegant solution. Thanks so much for the idea! :)